File: leetcode/algorithms/783_minimum_distance_between_bst_nodes/main.js
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

/**
 * @param {TreeNode} node
 * @param {number[]} vals
 */
var inorderTraversal = function (node, vals) {
  if (!node) {
    return;
  }

  inorderTraversal(node.left, vals);
  vals.push(node.val);
  inorderTraversal(node.right, vals);
};

/**
 * @param {TreeNode} root
 * @return {number}
 */
var minDiffInBST = function (root) {
  const vals = Array();

  inorderTraversal(root, vals);

  let min = 10e5;

  for (let i = 0; i < vals.length - 1; i++) {
    const diff = vals[i + 1] - vals[i];
    if (diff < min) {
      min = diff;
    }
  }

  return min;
};

File: leetcode/algorithms/783_minimum_distance_between_bst_nodes/main.py
from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class MinimumDistanceBetweenBSTNodes:
    def inorderTraversal(self, node: Optional[TreeNode], vals: List[int]):
        if node == None:
            return

        self.inorderTraversal(node=node.left, vals=vals)
        vals.append(node.val)
        self.inorderTraversal(node=node.right, vals=vals)

    def minDiffInBST(self, root: Optional[TreeNode]) -> int:
        vals: List[int] = []

        self.inorderTraversal(node=root, vals=vals)

        min: int = 10e5

        for i in range(0, len(vals) - 1):
            diff: int = vals[i + 1] - vals[i]
            if diff < min:
                min = diff

        return min


    # Solution
    pre = -float("inf")
    res = float("inf")

    def solution(self, root: Optional[TreeNode]) -> int:
        if root.left:
            self.minDiffInBST(root.left)

        self.res = min(self.res, root.val - self.pre)
        self.pre = root.val

        if root.right:
            self.minDiffInBST(root.right)

        return self.res
